mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,225 @@
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials)
+            service = BlobClient.from_blob_url(blob_url=credentials.signed_uri, credential=None,
+                                               headers=headers)
+            uploading_block_list = list()
+            for chunk in yield_file_in_chunks(local_file, _AZURE_MAX_BLOCK_CHUNK_SIZE):
+                block_id = base64.b64encode(uuid.uuid4().hex.encode())
+                try:
+                    service.stage_block(block_id, chunk, headers=headers)
+                except ClientAuthenticationError:
+                    _logger.warning(
+                        "Failed to authorize request, possibly due to credential expiration."
+                        "Refreshing credentials and trying again..")
+                    credentials = self._get_write_credentials(self.run_id,
+                                                              artifact_path).credentials.signed_uri
+                    service = BlobClient.from_blob_url(blob_url=credentials, credential=None)
+                    service.stage_block(block_id, chunk, headers=headers)
+                uploading_block_list.append(block_id)
+            try:
+                service.commit_block_list(uploading_block_list)
+            except ClientAuthenticationError:
+                _logger.warning(
+                    "Failed to authorize request, possibly due to credential expiration."
+                    "Refreshing credentials and trying again..")
+                credentials = self._get_write_credentials(self.run_id,
+                                                          artifact_path).credentials.signed_uri
+                service = BlobClient.from_blob_url(blob_url=credentials, credential=None)
+                service.commit_block_list(uploading_block_list)
+        except Exception as err:
+            raise MlflowException(err)
+
+    def _aws_upload_file(self, credentials, local_file):
+        try:
+            headers = self._extract_headers_from_credentials(credentials)
+            signed_write_uri = credentials.signed_uri
+            with open(local_file, 'rb') as file:
+                put_request = requests.put(signed_write_uri, headers=headers, data=file)
+                put_request.raise_for_status()
+        except Exception as err:
+            raise MlflowException(err)
+
+    def _upload_to_cloud(self, cloud_credentials, local_file, artifact_path):
+        if cloud_credentials.credentials.type == ArtifactCredentialType.AZURE_SAS_URI:
+            self._azure_upload_file(cloud_credentials.credentials, local_file, artifact_path)
+        elif cloud_credentials.credentials.type == ArtifactCredentialType.AWS_PRESIGNED_URL:
+            self._aws_upload_file(cloud_credentials.credentials, local_file)
+        else:
+            raise MlflowException('Not implemented yet')
```

Comment:
As we do in `_download_from_cloud`, we should use `Cloud provider not supported` in the exception text.

Reply:
Done.